[CI] Netty4IncrementalRequestHandlingIT testOversizedChunkedEncoding failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #5930 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/d3k6niyna3qok)
- [elasticsearch-periodic-platform-support #5920 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/cjuumemy4ssls)
- [elasticsearch-periodic-platform-support #5915 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/qzzhkm3opqvqo)
- [elasticsearch-periodic-platform-support #5915 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/hvahe2v5ksvmq)
- [elasticsearch-periodic-platform-support #5915 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/4yvc6krjzk73e)
- [elasticsearch-periodic-platform-support #5905 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/efaygairty4j2)
- [elasticsearch-periodic-platform-support #5905 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/h3nw7iwj5bdvk)
- [elasticsearch-periodic #5910 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/evunsifizeikw)
- [elasticsearch-periodic-platform-support #5900 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/g2fzumxsjejra)
- [elasticsearch-periodic-platform-support #5900 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/yuggwmrbug47a)

**Reproduction Line:**
```
./gradlew ":modules:transport-netty4:pooledInternalClusterTest" --tests "org.elasticsearch.http.netty4.Netty4IncrementalRequestHandlingIT.testOversizedChunkedEncoding" -Dtests.seed=719D1FB9F6972461 -Dtests.locale=es-PY -Dtests.timezone=Asia/Calcutta -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.http.netty4.Netty4IncrementalRequestHandlingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testOversizedChunkedEncoding),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: cannot request next chunk on closing stream
```

**Issue Reasons:**
- [8.x] 13 failures in test testOversizedChunkedEncoding (2.5% fail rate in 510 executions)
- [8.x] 2 failures in step rocky-8_platform-support-unix (28.6% fail rate in 7 executions)
- [8.x] 2 failures in step sles-12_platform-support-unix (28.6% fail rate in 7 executions)
- [8.x] 4 failures in step rhel-7_platform-support-unix (57.1% fail rate in 7 executions)
- [8.x] 6 failures in pipeline elasticsearch-periodic-platform-support (75.0% fail rate in 8 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequestBodyStream.java
+++ b/modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequestBodyStream.java
@@ -78,12 +78,12 @@ public class Netty4HttpRequestBodyStream implements HttpBody.Stream {
         assert handler != null : "handler must be set before requesting next chunk";
         requestContext = threadContext.newStoredContext();
         channel.eventLoop().submit(() -> {
-            if (closing) {
-                return;
-            }
             activityTracker.startActivity();
             requested = true;
             try {
+                if (closing) {
+                    return;
+                }
                 if (buf == null) {
                     channel.read();
                 } else {

--- a/modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequestBodyStream.java
+++ b/modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequestBodyStream.java
@@ -75,10 +75,12 @@ public class Netty4HttpRequestBodyStream implements HttpBody.Stream {
 
     @Override
     public void next() {
-        assert closing == false : "cannot request next chunk on closing stream";
         assert handler != null : "handler must be set before requesting next chunk";
         requestContext = threadContext.newStoredContext();
         channel.eventLoop().submit(() -> {
+            if (closing) {
+                return;
+            }
             activityTracker.startActivity();
             requested = true;
             try {

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -244,9 +244,6 @@ tests:
 - class: org.elasticsearch.search.ccs.CrossClusterIT
   method: testCancel
   issue: https://github.com/elastic/elasticsearch/issues/108061
-- class: org.elasticsearch.http.netty4.Netty4IncrementalRequestHandlingIT
-  method: testOversizedChunkedEncoding
-  issue: https://github.com/elastic/elasticsearch/issues/120444
 - class: org.elasticsearch.xpack.logsdb.seqno.RetentionLeaseRestIT
   issue: https://github.com/elastic/elasticsearch/issues/120434
 

